lib/pdk/config/namespace.rb

```diff
@@ -98,6 +98,8 @@ def [](key)
         return nil if settings[key.to_s].nil?
         return settings[key.to_s].value unless settings[key.to_s].value.nil?
         default_value = settings[key.to_s].default
+        # Duplicate arrays and hashes so that they are isolated from changes being made
+        default_value = default_value.dup if default_value.is_a?(Hash) || default_value.is_a?(Array)
```

Comment:
Are these still shallow copies? Is there a risk if there are nested hashes or arrays?

Reply:
Whoops. forgot to comment.  Created a deep_dup method to do the copying.